cnf-certification-test/accesscontrol/suite.go

```diff
@@ -303,3 +322,132 @@ func TestPodServiceAccount(env *provider.TestEnvironment) {
+		if put.Spec.ServiceAccountName == "" {
+			ginkgo.Skip("Can not test when serviceAccountName is empty. Please check previous tests for failures")
+		}
+
+		// Create a new object with the ability to gather clusterrolebinding specs.
+		rbTester := rbac.NewClusterRoleBindingTester(put.Spec.ServiceAccountName, put.Namespace, clientsholder.NewClientsHolder())
+
+		// Get any clusterrolebindings that do not belong to the pod namespace.
+		clusterRoleBindings, err := rbTester.GetClusterRoleBindings()
+		if err != nil {
+			failedPods = append(failedPods, put.Name)
+		}
+
+		if len(clusterRoleBindings) > 0 {
+			logrus.Warnf("Pod: %s has the following cluster role bindings: %s", put.Name, clusterRoleBindings)
+			failedPods = append(failedPods, put.Name)
+		}
+	}
+	if n := len(failedPods); n > 0 {
+		logrus.Debugf("Pods with cluster role bindings: %+v", failedPods)
+		ginkgo.Fail(fmt.Sprintf("%d pods have cluster role bindings in other namespaces.", n))
+	}
+}
+
+//nolint:funlen
+func TestAutomountServiceToken(env *provider.TestEnvironment) {
+	ginkgo.By("Should have automountServiceAccountToken set to false")
+
+	msg := []string{}
+	failedPods := []string{}
+	for _, put := range env.Pods {
+		ginkgo.By(fmt.Sprintf("check the existence of pod service account %s (ns= %s )", put.Namespace, put.Name))
+		gomega.Expect(put.Spec.ServiceAccountName).ToNot(gomega.BeEmpty())
+
+		// The token can be specified in the pod directly
+		// or it can be specified in the service account of the pod
+		// if no service account is configured, then the pod will use the configuration
+		// of the default service account in that namespace
+		// the token defined in the pod has takes precedence
+		// the test would pass iif token is explicitly set to false
+		// if the token is set to true in the pod, the test would fail right away
+		podServiceTokenStatus := rbac.DetermineStatus(put.Spec.AutomountServiceAccountToken)
+		if podServiceTokenStatus.TokenValue {
+			msg = append(msg, fmt.Sprintf("Pod %s:%s is configured with automountServiceAccountToken set to true ", put.Namespace, put.Name))
+			failedPods = append(failedPods, put.Name)
+			continue
+		}
+
+		// Collect information about the service account attached to the pod.
+		crbTester := rbac.NewAutomountTester(put.Spec.ServiceAccountName, put.Namespace, clientsholder.NewClientsHolder())
+		saStatus, err := crbTester.AutomountServiceAccountSetOnSA()
+		if err != nil {
+			failedPods = append(failedPods, put.Name)
+			continue
+		}
+
+		// The pod token is false means the pod is configured properly
+		// The pod is not configured and the service account is configured with false means
+		// the pod will inherit the behavior `false` and the test would pass
+		if !podServiceTokenStatus.TokenValue && !saStatus.TokenValue {
+			continue
+		}
+
+		// the service account is configured with true means all the pods
+		// using this service account are not configured properly, register the error
```

Comment:
// the service account is configured with true means not all the pods
		// using this service account are configured properly, register the error
		// message and fail